onnxruntime/core/graph/graph_utils.cc

```diff
@@ -353,12 +353,37 @@ const ONNX_NAMESPACE::AttributeProto* GetNodeAttribute(const Node& node, const s
   return iter == attrs.end() ? nullptr : &iter->second;
 }
 
+bool IsSingleOutputUsed(const Node& node) {
+  // TODO: check subgraphs.
+  std::vector<GraphEdge> output_edges = GetNodeOutputEdges(node);
```

Comment:
I don't think anything extra is needed for subgraphs given you're determining whether an output is used or not. There will be an output edge between this node and a control flow node with a subgraph for any implicit inputs to the subgraph, so iterating the output edges should be enough.

Reply:
This would also apply to the `IsOutputUsed` method right?

Reply:
Correct

---
In reply to: [284017081](https://github.com/microsoft/onnxruntime/pull/1018#discussion_r284017081) [](ancestors = 284017081)